docs/high-availability.md

```diff
@@ -0,0 +1,70 @@
+# High Availability with s3gw and Longhorn
+
+The s3gw mainly relies on Kubernetes and Longhorn to achieve High Availability (HA).
+
+A single s3gw instance is attached to its private Longhorn-provided persistent
+volume (PV) with Kubernetes restarting/redeploying as necessary on failure.
+
+In the context of the s3gw and Longhorn, this HA model is called: *Active/Standby*.
+
+The PV's state is guaranteed by Longhorn ensuring that a PV, along with
+its content, will always be available on all cluster nodes.
+This means that the s3gw can mount its Longhorn PV regardless of the
+current scheduling on the cluster.
+
+Obviously, the s3gw can't achieve higher availability or reliability than
+the underlying Longhorn volumes.
+This means that any corruption should occur in the file system is outside what
+s3gw can reasonably protect against; that's all undefined behavior and "restore
+from backup" time.
```

Comment:
I think we can drop the `that's all undefined behavior ...`, or maybe replace it with just `In this case, restoring from backup might be the only option.`.